### ISSUE TITLE
Possible memory leak

### ISSUE THREAD

**Issue by kadircet** [urllib3/urllib3#1514]
Hello everyone, thanks for the great work!

I was trying to update our internal version of urllib3 from 1.15.1 to 1.24.1 when running regression tests I've noticed there was an increase in memory usage for [apache/beam](https://github.com/apache/beam/blob/109eeb81535a1fd4f5d8cc724f3ddbddab326c3d/sdks/python/apache_beam/pipeline_test.py#L264) project. The memory usage for the test was increasing from ~300MB to ~450MB. Unfortunately I couldn't find the exact reason because project actually depends on requests which then depends on urllib3.

I was just wondering if anyone has noticed any regressions like this or has any pointers on debugging this one.

Have a nice day.

**Comment by sethmlarson** [comment]
Thanks for opening this. Could you point to where you're looking at in your CI to make the determination about the amount of tuples being used?

**Comment by kadircet** [comment]
Unfortunately I don't have any public links that I can provide since all tests are run in an internal system with authentication. But the process I followed is like this:

I've added two objgraph.show_growth() calls to the above mentioned test, one just before [pipeline.run()](https://github.com/apache/beam/blob/109eeb81535a1fd4f5d8cc724f3ddbddab326c3d/sdks/python/apache_beam/pipeline_test.py#L308) and another just after the [memory_usage](https://github.com/apache/beam/blob/109eeb81535a1fd4f5d8cc724f3ddbddab326c3d/sdks/python/apache_beam/pipeline_test.py#L281) then run first on the CI which gave ~600+(num_elements * num_maps) increase in number of tuples whereas my normal run only gave ~600 increase in number of tuples and the rest of the deltas seems similar.

I am sorry it is most likely not a good amount of information but I am not exactly a python guy :D I was just trying to update urllib3 for some other tool that needed it and in return faced with that problem.
